test/api/model/model_test.dart

```diff
@@ -212,4 +215,97 @@ void main() {
         .deepEquals([2, 3, 11]);
     });
   });
+
+  group('MessageEditState', () {
+    Map<String, dynamic> baseJson() => deepToJson(eg.streamMessage()) as Map<String, dynamic>;
+
+    test('edit history is absent', () {
+      check(Message.fromJson(baseJson()..['edit_history'] = null))
+        .editState.equals(MessageEditState.none);
+
+      check(Message.fromJson(baseJson()
+          ..['edit_history'] = null
+          ..['last_edit_timestamp'] = 1678139636))
+        .editState.equals(MessageEditState.edited);
+    });
+
+    void checkEditState(MessageEditState editState, List<Map<String, dynamic>> editHistory){
+      check(Message.fromJson(baseJson()..['edit_history'] = editHistory))
+        .editState.equals(editState);
+    }
+
+    group('edit history exists', () {
+      test('Channel name change only -> moved', () {
+        checkEditState(MessageEditState.moved, [{
+          'prev_stream': 'old_stream',
+          'stream': 'new_stream',
```

Comment:
These are meant to be channel IDs, right? Let's use integers, for a more realistic representation.